d2compiler/compile.go

```diff
@@ -116,15 +116,26 @@ func (c *compiler) compileBoardsField(g *d2graph.Graph, ir *d2ir.Map, fieldName
 }
 
 type compiler struct {
-	inEdgeGroup bool
-	err         d2parser.ParseError
+	err d2parser.ParseError
 }
 
 func (c *compiler) errorf(n d2ast.Node, f string, v ...interface{}) {
 	c.err.Errors = append(c.err.Errors, d2parser.Errorf(n, f, v...).(d2ast.Error))
 }
 
 func (c *compiler) compileMap(obj *d2graph.Object, m *d2ir.Map) {
+	class := m.GetField("class")
+	if class != nil {
+		className := class.Primary()
+		if className == nil {
+			c.errorf(class.LastRef().AST(), "class missing value")
+		} else {
+			classMap := m.GetClosestClassMap(className.String())
+			if classMap != nil {
+				c.compileMap(obj, classMap)
```

Comment:
This isn't ideal. If you make a change to the class after assigning the class to a shape, I'd expect the adjusted class to apply to the shape. I think we should use overlay instead here and overlay shape maps onto their classes at the very end.

Reply:
oh good call, will add a test for that

Reply:
i thought this would fail, but this actually passes:
https://github.com/terrastruct/d2/pull/772/commits/5dd273da18edf9b53be8c9165e5b171b0f6dbcc9

i guess d2ir must already merge

Reply:
Yea makes sense. I didn't realize some of the code changes were in `d2compiler` for some reason.